src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.TrackingSession.cs

```diff
@@ -96,17 +100,18 @@ private void QueueUpdateToStateMachine(StateMachine stateMachine, Task task)
             {
                 var asyncToken = _asyncListener.BeginAsyncOperation($"{GetType().Name}.{nameof(QueueUpdateToStateMachine)}");
 
-                task.SafeContinueWith(t =>
+                task.SafeContinueWithFromAsync(async t =>
                    {
-                       AssertIsForeground();
+                       await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_cancellationToken);
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.